// Copyright (C) 2023 Jose Mendoza. All rights reserved.
// Use of this source code is governed by an MIT license that can
// be found in the LICENSE file.

var mut runtime_is_started := false;

func assert(cond: bool, msg: string) {
    if !cond {
        console_ewriteln("panic: assertion failed: {}", msg);
        bt_print(2);
        process_exit(1);
    }
}

func runtime_error(s: string, args: ...Stringable) -> never {
    console_ewriteln("runtime error: {}", s.fmt(args));
    bt_print(2);
    process_exit(100);
}
